File: python/example2/vector2.py
class Vector2:
    def __init__(self, x ,y):
        self.x = x
        self.y = y

    def __add__(self, rhs):
        return Vector2(self.x + rhs.x, self.y + rhs.y)

    def __sub__(self, rhs):
        return Vector2(self.x - rhs.x, self.y - rhs.y)

    def __eq__(self, rhs):
        return (self.x == rhs.x and self.y == rhs.y)

File: cpp/example2/vector2.cpp
class Vector2
{
    public:
        Vector2(int x = 0, int y = 0) : x(x), y(y) {}

        Vector2 operator+(const Vector2& rhs) const
        {
            return Vector2(x + rhs.x, y + rhs.y);
        }

        Vector2 operator-(const Vector2& rhs) const
        {
            return Vector2(x - rhs.x, y - rhs.y);
        }

        bool operator==(const Vector2& rhs) const
        {
            return x == rhs.x && y == rhs.y;
        }

    private:
        int x;
        int y;
};
